### PR TITLE
Further supervision fixes.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

File: arangod/Agency/Job.cpp [hunk 1]
```diff
@@ -97,37 +97,66 @@ bool Job::finish(std::string const& server, std::string const& shard,
...
+  // Additional payload, which is to be executed in the finish transaction
+  Slice operations, preconditions;
+  
+  if (payload != nullptr) {
+    Slice slice = payload->slice();
+    TRI_ASSERT(slice.isObject() || slice.isArray());
+    if (slice.isObject()) {     // opers only
+      operations = slice;
+    } else {
+      TRI_ASSERT(slice.length() < 3); // opers + precs only
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Agency/Job.cpp]: Read further above?

File: arangod/Agency/MoveShard.cpp [hunk 2]
```diff
@@ -746,9 +752,24 @@ arangodb::Result MoveShard::abort() {
...
-    finish("", "", true, "job aborted");
+
+    // Do NOT remove, just cause it seems obvious!
+    // We're working off a snapshot.
+    // Make sure ToDo is still actually to be done
+    auto todoPrec = std::make_shared<Builder>();
+    { VPackArrayBuilder b(todoPrec.get());
+      { VPackObjectBuilder o(todoPrec.get()); } // nothing to declare
+      { VPackObjectBuilder path(todoPrec.get()); // expect jobs still to be sitting in ToDo
+        VPackValue(toDoPrefix + _jobId);
```
Comment (neunhoef) [inline arangod/Agency/MoveShard.cpp]: Why does this compile in the first place? I think it needs to be `todoPrec->add(VPackValue(toDoPrefix + _jobId));`.

File: arangod/Agency/MoveShard.cpp [hunk 3]
```diff
@@ -746,9 +752,24 @@ arangodb::Result MoveShard::abort() {
...
+        {
```
Comment (neunhoef) [inline arangod/Agency/MoveShard.cpp]: THese braces are not necessary.

Comment (neunhoef) [reply arangod/Agency/MoveShard.cpp]: I mean, another VPackObjectBuilder is needed.

**Comment by kvahed** [comment]
Approved

**Pull request merged**